x/twap/migrate.go

```diff
@@ -14,3 +32,46 @@ func (k Keeper) MigrateExistingPools(ctx sdk.Context, latestPoolId uint64) error
 	}
 	return nil
 }
+
+// initializeGeometricTwapAcc initializes all existing twap records in state
+// with the given value. Returns nil on success, error otherwise.
+// Returns error if:
+// - fails to get historical time indexed records.
+// - does not find any time indexed records.
+// - the time indexed records returned from state are written back
+// in incorrect order. The order must be ascending by time so that
+// we update the "most recent" index with the right records correctly.
+func (k Keeper) initializeGeometricTwapAcc(ctx sdk.Context, value sdk.Dec) error {
+	// In ascending order by time.
+	historicalTimeIndexed, err := k.getAllHistoricalTimeIndexedTWAPs(ctx)
+	if err != nil {
+		return err
+	}
+
+	if len(historicalTimeIndexed) == 0 {
+		return errors.New("error: no historical twap records found")
+	}
+
+	// Since we are iterate over time-indexed records in ascending order,
+	// most recent record should also be updated correctly.
+	for i, record := range historicalTimeIndexed {
+		record := record
+		// Sanity check order.
+		if i > 0 {
+			previousRecord := historicalTimeIndexed[i-1]
+
+			isInvalidOrder := previousRecord.Time.After(record.Time)
+			if isInvalidOrder {
+				return fmt.Errorf("error: historical twap records are not in ascending order, (%v), was after (%v)", previousRecord, record)
+			}
+		}
```

Comment:
Do we have proof that `getAllHistoricalTimeIndexedTWAPs` returns in correctly in ascending order? Looked into existing tests and could not find any relevent test